internal/vault/client.go

```diff
@@ -32,6 +32,9 @@ const (
 
 // Client holds api.Client and handles state required to renew tokens and re-auth as required.
 type Client struct {
+	Name    string
+	Version string
```

Comment:
Looks like they are accessed from the `proxy` package. However, that is also the _only_ place they're accessed from. I can see how it was a bit awkward to access these constants because they're defined in the main package. Maybe we should have some exported `ExtensionName` and `ExtensionVersion` constants in the `config` package? Then every consumer can happily reference them directly from there.